packages/dd-trace/src/config.js

```diff
@@ -593,6 +593,7 @@ class Config {
     this._setValue(defaults, 'instrumentation_config_id', undefined)
     this._setValue(defaults, 'trace.aws.addSpanPointers', true)
     this._setValue(defaults, 'trace.dynamoDb.tablePrimaryKeys', undefined)
+    this._setValue(defaults, 'fetchAgentInfo', true)
```

Comment:
This should only be true by default in SSI as discussed. This is because in SSI we have better guarantees for what agent is installed where and with what version. In other environments, agents can be installed elsewhere, can change versions whenever, can be in the middle of a rolling update with multiple versions at the same time, etc.

Reply:
no longer relevant, we are just using a config for whether to encode span events top level